tests/unit/test_salesforce.py

```diff
@@ -0,0 +1,166 @@
+"""'test_salesforce.py'."""
+
+import pytest
+from simple_salesforce import Salesforce
+
+from viadot.exceptions import CredentialError
+from viadot.sources import SalesForce
+from viadot.sources.salesforce import SalesForceCredentials
+
+variables = {
+    "credentials": {
+        "username": "test_user",
+        "password": "test_password",
+        "token": "test_token",
+    },
+    "records_1": [
+        {
+            "Id": "001",
+            "Name": "Test Record",
+            "attributes": {
+                "type": "Account",
+                "url": "/services/data/v50.0/sobjects/Account/001",
+            },
+        },
+    ],
+    "records_2": [
+        {
+            "Id": "001",
+            "Name": "Test Record",
+            "attributes": {
+                "type": "Account",
+                "url": "/services/data/v50.0/sobjects/Account/001",
+            },
+        },
+    ],
+    "data": [
+        {"Id": "001", "Name": "Test Record"},
+        {"Id": "002", "Name": "Another Record"},
+    ],
+}
+
+
+@pytest.mark.basic()
+def test_salesforce_init_dev_env(mocker):
+    """Test SalesForce, starting in dev mode."""
+    mock_sf_instance = mocker.MagicMock(spec=Salesforce)
+    mocker.patch("viadot.sources.salesforce.Salesforce", return_value=mock_sf_instance)
```

Comment:
You repeat these lines a lot of times; this should be a fixture

Reply:
I have one day and a half to review all the PRs, so I would recommend, again, adding it to a backlog task.

Reply:
It's ok with me, pls paste the link here once done so we can close this

Reply:
backlog 2.0 issue:
#1055